### ISSUE TITLE
[Bug]: Reapplying transform from history adds black borders to the result image

### ISSUE THREAD

**Issue by efalco777** [hm21/pro_image_editor#120]
### Package Version

4.0.4

### Flutter Version

3.22.2

### Platforms

iOS

### How to reproduce?

Have an image for ex. 1000x1000 size
1. Crop the image to 1000x500
2. Save the image & export the history(the result generated image file is cropped correctly, result is 1000x500)
3. Reopen the original image with the previous exported history
4. Save the image again without applying any changes related to transform.

Expected:
Crop works the same way when reapplied from history

Actual:
Resulting image has the original 1000x1000 size. The actual cropped image is in the middle, remaining area is filled with black colour.

note:
Same black areas are applied when rotating the image from history that isn't an exact square.

### Device Model (optional)

iPhone 15

**Comment by hm21** [comment]
Thank you for reporting this issue.

I can't reproduce this problem either. But do I understand you correctly that when you import the history, you use the new image that the editor cropped (1000×500) and not the original (1000×1000)? If yes, then this case will not work because the editor wants to apply the crop transformation to an image with the old size (1000×1000) that you can continue where you exported the history. When you now import the new image size, the part is already cropped, so the editor has to fill it with a transparent area because the editor can't know how this part looked like before. Anyway, you now have two options.
#### Option 1
Export the history without the crop/rotate transform configs like below:
```dart
var export = await editor.exportStateHistory(
  configs: const ExportEditorConfigs(
    historySpan: ExportHistorySpan.current,
    exportCropRotate: false,
  ),
);
```

#### Option 2
Import the original image, that the user can continue where he exports the history, which also allows him to redo the crop/rotate transformations.

**Comment by efalco777** [comment]
I'm reusing the old file always. Here's a gif for the black borders:
![Simulator Screen Recording - iPhone 15 - 2024-06-20 at 11 04 18](https://github.com/hm21/pro_image_editor/assets/20524697/4f68eba8-7655-4719-b9d6-02de57d2cd1f)

This is the absolute minimal example i could come up with that's used in the gif:

```
class _FileEditor extends StatefulWidget {
  const _FileEditor({
    required this.file,
  });

  final File file;

  @override
  State<_FileEditor> createState() => _FileEditorState();
}

class _FileEditorState extends State<_FileEditor> {
  late GlobalKey<ProImageEditorState> _key;
  String? _historyJson;

  Uint8List? _editedPhotoBytes;
  bool _showEditor = true;

  @override
  void initState() {
    super.initState();
    _key = GlobalKey<ProImageEditorState>();
  }

  @override
  Widget build(BuildContext context) {
    return switch (_showEditor) {
      true => ProImageEditor.file(
          widget.file,
          key: _key,
          callbacks: ProImageEditorCallbacks(
            onImageEditingComplete: onImageEditingComplete,
          ),
          configs: ProImageEditorConfigs(
            stickerEditorConfigs: StickerEditorConfigs(
              enabled: true,
              buildStickers: (setLayer, scrollController) {
                return SizedBox(
                  height: 200,
                  width: double.infinity,
                  child: ElevatedButton(
                    child: const Text('Create sticker'),
                    onPressed: () {
                      final sticker = Container(
                        color: Colors.red,
                        width: 50,
                        height: 50,
                      );
                      setLayer(sticker);
                    },
                  ),
                );
              },
            ),
            stateHistoryConfigs: StateHistoryConfigs(
              initStateHistory: _historyJson != null ? ImportStateHistory.fromJson(_historyJson!) : null,
            ),
          ),
        ),
      false => Scaffold(
          body: Column(
            children: [
              SizedBox(
                height: 400,
                child: Image.memory(_editedPhotoBytes!),
              ),
              const Divider(),
              ElevatedButton(
                onPressed: () => _openEditor(),
                child: const Text('Reopen editor with the history'),
              )
            ],
          ),
        ),
    };
  }

  Future<void> onImageEditingComplete(bytes) async {
    final exportState = await _key.currentState?.exportStateHistory();
    _closeEditor(await exportState?.toJson(), bytes);
  }

  void _closeEditor(String? historyJson, Uint8List? bytes) {
    setState(() {
      _showEditor = false;
      _historyJson = historyJson;
      _editedPhotoBytes = bytes;
    });
  }

  void _openEditor() {
    print('Reopening with: ' + (_historyJson?.toString() ?? 'empty'));
    setState(() {
      _key = GlobalKey<ProImageEditorState>();
      _showEditor = true;
    });
  }
}
```

**Comment by efalco777** [comment]
Reposting the gif:
![Screenshot 2024-06-19 at 13 46 18](https://github.com/hm21/pro_image_editor/assets/20524697/f66267ca-71df-4646-b11e-793cf4278b44)

**Comment by efalco777** [comment]
Reposting gif as video since it seems like gif drops the important parts:
https://github.com/hm21/pro_image_editor/assets/20524697/56838483-df7a-4873-965c-37e853bc5c0f

**Comment by hm21** [comment]
Thank you so much for the example, it helped me a lot to reproduce your issue. I released now version `4.0.8` which should resolve it. Please let me know when the issue persists by you after you update.
